src/Service/Resolvers/BaseQueryStructure.cs

```diff
@@ -109,11 +111,19 @@ public class BaseQueryStructure
         ///  Add parameter to Parameters and return the name associated with it
         /// </summary>
         /// <param name="value">Value to be assigned to parameter, which can be null for nullable columns.</param>
-        public string MakeParamWithValue(object? value)
+        public virtual string MakeParamWithValue(object? value, string? paramName = null)
```

Comment:
Too many paramName variables floating here: 

1. columnName
2. storedprocedure param name
3. ConnectionParam name - sent as part of the dbcommand 
Can you please add comments/rename some of them to make it clearer?